### PR TITLE
Use Black to format Python files

### PR DESCRIPTION
**Pull request by kuhar**
Switch from yapf to Black to better align with the LLVM and broader Python community. I decided not to go with Pyink as it seems much less popular and differs in formatting style beyond indentation. 

-  Reformat all python files outside of `third_party` with black.
-  Update the lint workflow to use black. This only considers files modified by the PR.
-  Delete old dotfiles.

The command used to reformat all files at once:
```shell
fd -e py --exclude third_party | xargs black
```

To learn more about Back, see: https://black.readthedocs.io/en/stable/ and https://github.com/psf/black.

In the next PR, once the commit SHA of this PR is finalized, I plan to add this commit to `.git-blame-ignore-revs` to keep the blame history clean.

Issue: https://github.com/openxla/iree/issues/14135

### CODE REVIEW

**Comment by kuhar** [comment]
Hmm, any idea what the pytype check complains about?

**Comment by ScottTodd** [comment]
> Hmm, any idea what the pytype check complains about?

`ninja: error: build.ninja:133: multiple rules generate /home/runner/work/iree/iree/.pytype/pyi/__main__.pyi`
We've seen that `multiple rules generate __main__.py` for as long as we've been using pytype. The check is non-blocking because of it.

**Comment by kuhar** [comment]
It also seems like yapf was marked as a required workflow but if absent now. I think this will have to be changed through the web ui.

File: .github/workflows/lint.yml
Comment (GMNGeoffrey) [inline .github/workflows/lint.yml:1]: I think there's a typo in the PR description: "LLVM and Python Python community."

File: build_tools/bazel_to_cmake/bazel_to_cmake.py [hunk 1]
```diff
@@ -69,274 +69,299 @@
...
+        " 2: Also output when conversion was successful.",
+    )
+
+    # Specify only one of these (defaults to --root_dir=<main source dirs>).
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--dir", help="Converts the BUILD file in the given directory", default=None
+    )
+    default_root_dirs = (
+        repo_cfg.DEFAULT_ROOT_DIRS if hasattr(repo_cfg, "DEFAULT_ROOT_DIRS") else []
```
Comment (GMNGeoffrey) [inline build_tools/bazel_to_cmake/bazel_to_cmake.py:111]: Well that one's much nicer!

File: .github/workflows/lint.yml
Comment (GMNGeoffrey) [inline .github/workflows/lint.yml:1]: Can you include in the description the actual commands used to perform this update? It's obviously not really feasible to review all the changes, but would be good to know if it was as simple as `black -r .` or whether you had to make manual adjustments or what

Comment (GMNGeoffrey) [inline .github/workflows/lint.yml:1]: I've removed the yapf job as a required check

Comment (kuhar) [reply .github/workflows/lint.yml:1]: yep, thanks

Comment (kuhar) [reply .github/workflows/lint.yml:1]: Good idea, done

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ jobs:
           git diff --exit-code
           exit "${EXIT_CODE?}"
 
-  yapf:
+  black:
```
Comment (ScottTodd) [inline .github/workflows/lint.yml:52]: Can you add a link to https://black.readthedocs.io/en/stable/ and/or https://pypi.org/project/black/ in the PR description?

**Review by ScottTodd — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 2 above)
Comment (kuhar) [reply .github/workflows/lint.yml:52]: Good idea, done

**Pull request merged**